torch/csrc/jit/python/pybind_utils.cpp

```diff
@@ -641,7 +641,11 @@ py::object toPyObject(IValue ivalue) {
     for (const auto i : c10::irange(list.size())) {
       t[i] = toPyObject(IValue{list.get(i)});
     }
+#if __cplusplus >= 202002L
```

Comment:
Do you have details on what this is about?
We don't use this pattern more often?

Reply:
g++9 doesn't support return without std::move here. However, if C++20 is enabled on newer g++, it reports redundant move errors.

Reply:
Do we expect more of these to be around? If so, let's make a macro!

Reply:
Finally we should keep the newer version when the old compilers are no longer supported, which is likely to happen this year IMO.

Reply:
It would be good to clarify, whether you are trying to workaround g++-9 implementation bug or something fundamental that were added in C++20 standard. As such, imo this check should be for a specific compiler version, and I agree with @albanD it should be `MOVE_UNLESS_GCC_9` macro